pulp_container/app/registry_api.py

```diff
@@ -211,10 +212,21 @@ class ContainerRegistryApiMixin:
     It sets token authentication and token permission.
     """
 
-    authentication_classes = [TokenAuthentication]
     permission_classes = [TokenPermission]
     schema = None
 
+    @property
+    def authentication_classes(self):
+        if settings.get("TOKEN_AUTH_DISABLED", False):
+            return api_settings.DEFAULT_AUTHENTICATION_CLASSES
```

Comment:
I decided that this is the better way, as we can fallback to the _list_ of configured auth-classes.
Also it does not mess with the class creation process as `__new__` does.

Reply:
this changed fixed the direct api calls but still not for client push
`return [api_settings.DEFAULT_AUTHENTICATION_CLASSES[1]]`

Reply:
i get this `"Attempting next endpoint for push after error: unauthorized: Authentication credentials were not provided."` 
it feels like the client does not send the basic auth creds in the calls?

Reply:
I used podman, and push with logged in as admin worked for me.

Reply:
podman-2.2.1-1.fc32.x86_64